calculator.html

```diff
@@ -183,40 +183,45 @@
         numCheck  = 0
       }
     }
-      
+
     function equals(){
       var val = parseFloat(digit.join(''))
-      var value   = val
-      if(signCheck==4) {
-        var result = storage * value
-        console.log(storage + " x " + value + " = " + result)           
-        storage    = result
-        console.log ("the stored value is now " + storage)
-      }
-      if(signCheck==3) {
-        if(value!=0){
-        var result = storage / value
-        console.log(storage + " / " + value + " = " + result)          
-        storage    = result
-        console.log ("the stored value is now " + storage)
-        }
-        else{
-          console.log("can't divide by 0!")
-        }
-      }
-      if(signCheck==2) {
-        var result = storage - value
-        console.log(storage + " - " + value + " = " + result)
-        storage = result
-        console.log ("the stored value is now " + storage)
-      }
-      if(signCheck==1) {
-        var result = storage + value
-        console.log(storage + " + " + value + " = " + result)
-        storage    = result
-        console.log ("the stored value is now " + storage)
+      var value = val
+      var result = null;
+      switch(signCheck) {
+        case 4:
+          result = storage * value
+          console.log(storage + " x " + value + " = " + result)           
+          storage    = result
+          console.log ("the stored value is now " + storage)
+          break;
+        case 3:
+          if(value!=0){
+          result = storage / value
+          console.log(storage + " / " + value + " = " + result)          
+          storage    = result
+          console.log ("the stored value is now " + storage)
+          }
+          else{
+            console.log("can't divide by 0!")
+          }
+          break;
+        case 2:
+          result = storage - value
+          console.log(storage + " - " + value + " = " + result)
+          storage = result
+          console.log ("the stored value is now " + storage)
+          break;
+        case 1:
+          result = storage + value
+          console.log(storage + " + " + value + " = " + result)
+          storage    = result
+          console.log ("the stored value is now " + storage)
+          break;
+        default:
+          console.log("No operation or an unknown operation was requested")
       }
-      if(result!==undefined){
+      if(result){
```

Comment:
Rather than compare directly with 'undefined', most languages have the concept of coersing truthy or falsy values out of things. More here: https://developer.mozilla.org/en-US/docs/Glossary/Truthy 

the reason is basically a language needs to privide opinions over these things: if a var is defined but empty, should a presence test on the var eval to true or false? each language differs but they all have coersive constructs. Here i'm using `if (result)` and letting JS tell me if i have a result based on those opinions.